src/Core/Framework/Api/ApiException.php

```diff
@@ -322,10 +322,23 @@ public static function invalidSchemaDefinitions(string $filename, \JsonException
         );
     }
 
+    /**
+     * @deprecated tag:v6.7.0 - Will return status code 400 instead of 500
+     */
     public static function invalidAccessKey(): self
     {
+        if (!Feature::isActive('v6.7.0.0')) {
+            Feature::triggerDeprecationOrThrow('v6.7.0.0', 'Will return status code 400 instead of 500');
+
+            return new self(
+                Response::HTTP_INTERNAL_SERVER_ERROR,
+                self::API_INVALID_ACCESS_KEY_EXCEPTION,
+                'Access key is invalid and could not be identified.',
+            );
+        }
+
         return new self(
-            Response::HTTP_INTERNAL_SERVER_ERROR,
+            Response::HTTP_BAD_REQUEST,
```

Comment:
To me `400 bad request` is still not quite what I would expect when I also get the error message `Access key is invalid and could not be identified.`.

I would actually expect a `403 Forbidden` here, see
https://en.wikipedia.org/wiki/HTTP_403

But I also don't know the details where this exception is raised. Is this "access key" provided by the http client? Or is it an internal thing we use to authenticate against a third party service? If it's the latter I would actually stay with `500` and adjust the error message to better reflect that.

Reply:
Unfortunately I can't comment everywhere in this file, but the function below `invalidAccessKeyIdentifier` sounds kinda similar and is also `500` right now. Maybe it makes sense to switch that to `400 bad request` 🤔